src/main/kotlin/no/nav/syfo/AccessTokenClient.kt

```diff
@@ -0,0 +1,41 @@
+package no.nav.syfo
+
+import io.ktor.client.HttpClient
+import io.ktor.client.request.accept
+import io.ktor.client.request.forms.FormDataContent
+import io.ktor.client.request.post
+import io.ktor.http.ContentType
+import io.ktor.http.HttpMethod
+import io.ktor.http.Parameters
+import org.slf4j.Logger
+import org.slf4j.LoggerFactory
+
+class AccessTokenClient(private val aadAccessTokenUrl: String, private val clientId: String, private val clientSecret: String, private val client: HttpClient) {
+    private val log: Logger = LoggerFactory.getLogger("no.nav.syfo.syfonarmesteleder")
+
+    suspend fun hentAccessToken(resource: String): String {
+        log.info("Henter token")
+        val response: AadAccessToken = client.post(aadAccessTokenUrl) {
+            accept(ContentType.Application.Json)
+            method = HttpMethod.Post
+            body = FormDataContent(Parameters.build {
+                append("client_id", clientId)
+                append("resource", resource)
+                append("grant_type", "client_credentials")
+                append("client_secret", clientSecret)
+            })
+        }
+        log.info("Har hentet accesstoken")
+        return response.access_token
+    }
+}
+
+private class AadAccessToken(
```

Comment:
Veldig fan av disse! Er det noen grunn til å ikke bruke data class? (Jeg vet ikke forskjellen for å være ærlig..)

Reply:
Jeg vet ikke forskjellen jeg heller, men jeg tror jeg mente at det skulle være en data class, jeg bare glemte å skrive det 😅

Reply:
Korrekt 🚀